### ISSUE TITLE
iasl crash when parsing

### ISSUE THREAD

**Issue by kraghot** [acpica/acpica#237]
I seem to get a crash when parsing my laptops DSDT file. The output is the following:
```
Intel ACPI Component Architecture
ASL+ Optimizing Compiler/Disassembler version 20170224
Copyright (c) 2000 - 2017 Intel Corporation

Input file dsdt/dsdt.dat, Length 0x19726 (104230) bytes
ACPI: DSDT 0x0000000000000000 019726 (v02 _ASUS_ Notebook 01072009 INTL 20120913)
Pass 1 parse of [DSDT]
Pass 2 parse of [DSDT]
Parsing Deferred Opcodes (Methods/Buffers/Packages/Regions)

Parsing completed

Found 2 external control methods, reparsing with new information
Pass 1 parse of [DSDT]
Pass 2 parse of [DSDT]
Parsing Deferred Opcodes (Methods/Buffers/Packages/Regions)

Parsing completed
*** Error in `iasl': munmap_chunk(): invalid pointer: 0x00000000004a76d5 ***
======= Backtrace: =========
/usr/lib/libc.so.6(+0x722ab)[0x7fc2c692f2ab]
/usr/lib/libc.so.6(+0x7890e)[0x7fc2c693590e]
iasl[0x43b09b]
iasl[0x43ac88]
iasl[0x43ad38]
iasl[0x437670]
iasl[0x41215b]
iasl[0x429017]
iasl[0x4292d4]
iasl[0x401e9b]
/usr/lib/libc.so.6(__libc_start_main+0xf1)[0x7fc2c68dd511]
iasl[0x401f4a]
======= Memory map: ========
00400000-004e1000 r-xp 00000000 b3:02 673194                             /usr/bin/iasl
006e0000-006e1000 r--p 000e0000 b3:02 673194                             /usr/bin/iasl
006e1000-006eb000 rw-p 000e1000 b3:02 673194                             /usr/bin/iasl
006eb000-00705000 rw-p 00000000 00:00 0 
00cff000-0131d000 rw-p 00000000 00:00 0                                  [heap]
7fc2c66a6000-7fc2c66bc000 r-xp 00000000 b3:02 656544                     /usr/lib/libgcc_s.so.1
7fc2c66bc000-7fc2c68bb000 ---p 00016000 b3:02 656544                     /usr/lib/libgcc_s.so.1
7fc2c68bb000-7fc2c68bc000 r--p 00015000 b3:02 656544                     /usr/lib/libgcc_s.so.1
7fc2c68bc000-7fc2c68bd000 rw-p 00016000 b3:02 656544                     /usr/lib/libgcc_s.so.1
7fc2c68bd000-7fc2c6a58000 r-xp 00000000 b3:02 654989                     /usr/lib/libc-2.25.so
7fc2c6a58000-7fc2c6c57000 ---p 0019b000 b3:02 654989                     /usr/lib/libc-2.25.so
7fc2c6c57000-7fc2c6c5b000 r--p 0019a000 b3:02 654989                     /usr/lib/libc-2.25.so
7fc2c6c5b000-7fc2c6c5d000 rw-p 0019e000 b3:02 654989                     /usr/lib/libc-2.25.so
7fc2c6c5d000-7fc2c6c61000 rw-p 00000000 00:00 0 
7fc2c6c61000-7fc2c6c84000 r-xp 00000000 b3:02 654990                     /usr/lib/ld-2.25.so
7fc2c6e53000-7fc2c6e55000 rw-p 00000000 00:00 0 
7fc2c6e82000-7fc2c6e83000 rw-p 00000000 00:00 0 
7fc2c6e83000-7fc2c6e84000 r--p 00022000 b3:02 654990                     /usr/lib/ld-2.25.so
7fc2c6e84000-7fc2c6e85000 rw-p 00023000 b3:02 654990                     /usr/lib/ld-2.25.so
7fc2c6e85000-7fc2c6e86000 rw-p 00000000 00:00 0 
7ffe4a475000-7ffe4a496000 rw-p 00000000 00:00 0                          [stack]
7ffe4a51f000-7ffe4a521000 r--p 00000000 00:00 0                          [vvar]
7ffe4a521000-7ffe4a523000 r-xp 00000000 00:00 0                          [vdso]
ffffffffff600000-ffffffffff601000 r-xp 00000000 00:00 0                  [vsyscall]
[1]    31795 abort (core dumped)  iasl -d dsdt/dsdt.dat ssdt/* -e other/*
```
The files I was trying to parse are the following:
[acpidump.tar.gz](https://github.com/acpica/acpica/files/852050/acpidump.tar.gz)

**Comment by debox1** [comment]
I can reproduce the issue on the R02_24_17 tag but on the latest R03_03_17 tag. Please try the latest.

**Comment by acpibob** [comment]
Yes, this should be fixed in the 20170303 ACPICA release.
